docs/sidebars.js

```diff
@@ -258,8 +255,11 @@ module.exports = {
       collapsed: true,
       items: [
         "nlu-based-assistants/glossary",
+        "concepts/components/custom-graph-components",
+        "concepts/components/llm-custom",        
+        "concepts/components/graph-recipe",
```

Comment:
for me it makes more sense to keep this under the Components section. Can you explain your reasoning?

Reply:
thinking is that components is under 'key concepts' which to me implies things you should understand as a Rasa user. IMO these pieces are not 'key' but rather advanced topics.

Reply:
My worry is that we put things under the Reference section and it becomes a catch-all section for everything that doesn't fit with Key concept or the sections we added.

Because one key selling point of Rasa is to allow for customisation and configuration, it could actually make sense to have a section for building advanced assistants for pushing the boundaries of what Rasa bakes in by default. WDYT? I would see that section between "Operating at scale" and "Reference".

Reply:
I like that idea, especially if we can use it as a place to articulate what we consider the public facing API and what things are internal and you shouldn't touch unless you want to make your own life hard. But clearly outside the scope of this PR. Happy to put this back in key concepts for now